From: Jiri Kosina <jkosina@suse.cz>
Subject: [PATCH] x86/bugs: make intel_rds_mask() honor X86_FEATURE_SSBD
References: bsc#1094019

Don't return SSBD mask from x86_calculate_kernel_spec_ctrl() if feature
bit is not set.

Signed-off-by: Jiri Kosina <jkosina@suse.cz>
---
 arch/x86/kernel/cpu/bugs.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/arch/x86/kernel/cpu/bugs.c
+++ b/arch/x86/kernel/cpu/bugs.c
@@ -158,6 +158,9 @@ static inline u64 intel_rds_mask(void)
 	if (boot_cpu_data.x86_vendor != X86_VENDOR_INTEL)
 		return 0;
 
+	if (!boot_cpu_has(X86_FEATURE_SSBD))
+		return 0;
+
 	mask = ssbd_tif_to_spec_ctrl(current_thread_info()->flags);
 
 	/*
